src/main/webapp/app/components/calendarButton/UsageAnalysisCalendarButton.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import {
+  InputGroup,
+  Button,
+  Dropdown,
+  DropdownButton,
+  Row,
+  Form,
+  Col,
+  Container,
+} from 'react-bootstrap';
+import classnames from 'classnames';
+import { getMomentInstance } from 'app/shared/utils/Utils';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import 'react-day-picker/lib/style.css';
+import moment from 'moment';
+import {
+  APP_LOCAL_DATETIME_FORMAT_Z_FORCE,
+  APP_LOCAL_DATE_FORMAT,
+} from 'app/config/constants';
+import MomentLocaleUtils from 'react-day-picker/moment';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+import pluralize from 'pluralize';
+
+type CalendarButtonProps = {
+  currentDate: string;
+  currentFromDate: string;
+  currentToDate: string;
+  fromDate: (newDate: string) => void;
+  toDate: (newDate: string) => void;
+  filterToggled: (filterActive: boolean) => void;
+};
+
+export const UsageAnalysisCalendarButton: React.FunctionComponent<CalendarButtonProps> = props => {
+  function handleFromDayChange(day: any) {
+    props.fromDate(day);
+    props.filterToggled(true);
+    if (props.currentToDate == null || props.currentToDate === 'Select Date') {
+      props.toDate(day);
+    }
+  }
+
+  function handleToDayChange(day: any) {
+    props.toDate(day);
+    props.filterToggled(true);
+  }
+
+  function handleResetDays() {
+    props.fromDate('Select Date');
+    props.toDate('Select Date');
+    props.filterToggled(false);
+  }
+
+  return (
+    <DefaultTooltip placement={'top'} overlay={'View Specific Date Range'}>
+      <DropdownButton
+        as={InputGroup.Append}
+        title={<i className={classnames('fa fa-calendar')}></i>}
+        id="time-select-dropdown"
+        className="ml-3"
+      >
+        <Container>
+          <div>
+            <Row>
+              <Col>From:</Col>
+            </Row>
+          </div>
+          <DayPickerInput
+            value={
+              props.currentFromDate === undefined ||
+              props.currentFromDate === 'Select Date'
```

Comment:
If the type defines as string and is required field, don't check whether it's undefined. If the expectation of `currentFromDate` is string || undefined, please update the type.